Update, Fixing bugs, added modal for project

TarikVelic opened the pull request:
--------------------------------------------------------------------
 MAIN
 * Changed from Last to Previous button in Pagination
 * Added Employee atribute instead of User in Table
 * Changed Footer of table so it can be added dynamically 
 * --------------------------------------------------------------------
 OTHER
 *
--------------------------------------------------------------------

TarikVelic committed:
Update, Fixing bugs, etc

--------------------------------------------------------------------
 MAIN
 * Changed from Last to Previous button in Pagination
 * Added Employee atribute instead of User in Table
 * Changed Footer of table so it can be added dynamically
--------------------------------------------------------------------
 OTHER
 *
--------------------------------------------------------------------

TarikVelic committed:
FEATURE/TariikMVP/Update
--------------------------------------------------------------------
 MAIN
 * Added Projects Modal
--------------------------------------------------------------------
 OTHER
 * <other msg>
--------------------------------------------------------------------

becarselman commented on src/components/_addProjectsModal/AddProjectsModal.js:
```diff
@@ -0,0 +1,162 @@
+    try {
+      actions.fetchAllEmployeesRequest();
+    } catch (error) {
+      console.error("Error fetching developers:", error);
```
please remove console.error. You can use antd notification or something similar. Also this need to be done in saga.

becarselman commented on src/components/_addProjectsModal/AddProjectsModal.js:
```diff
@@ -0,0 +1,162 @@
+  };
+
+  useEffect(() => {
+    setDeveloperOptions(
```
I think this is messy to use map inisde of setter. Can we separate this

becarselman commented on src/components/_addProjectsModal/AddProjectsModal.js:
```diff
@@ -0,0 +1,162 @@
+
+  const handleSubmit = async () => {
+    try {
+      const formattedDevelopers = developers.map((developerId) => {
```
I think I already see this block of code when you fetch developers for project table. If I am right then you can make helper function for this

becarselman replied:
and also try to avoid loop in loop because now you have n * n complexity. Sometimes https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce this can fix complexity

becarselman commented on src/components/_addProjectsModal/AddProjectsModal.js:
```diff
@@ -0,0 +1,162 @@
+      const formattedDevelopers = developers.map((developerId) => {
+        const selectedDeveloper = developerOptions.find((option) => option.value === developerId);
+        if (!selectedDeveloper) {
+          console.error(`Developer not found for developerId: ${developerId}`);
```
please remove console on all places

becarselman commented on src/components/_addProjectsModal/AddProjectsModal.js:
```diff
@@ -0,0 +1,162 @@
+        projectStatus,
+      };
+  
+      actions.createProjectRequest(projectData);
```
I don't like this. can we move logic as more as you can in saga.

becarselman commented on src/components/_addProjectsModal/Container.js:
I see you ask me can you connect here. You can and than you don't need to use index.js. But be careful  when you importing which file would you use.

TarikVelic committed:
Review comments fixes

becarselman commented on src/components/_addProjectsModal/AddProjectsModal.js:
```diff
@@ -0,0 +1,141 @@
+  const fetchDevelopers = async () => {
+    try {
+      actions.fetchAllEmployeesRequest();
+    } catch (error) {
```
why catch is empty

becarselman commented on src/components/_addProjectsModal/AddProjectsModal.js:
```diff
@@ -0,0 +1,141 @@
+    try {
+      const projectData = formatProjectData(name, description, duration, developers, hourlyRate, projectValue, projectStatus, developerOptions);
+      actions.createProjectRequest(projectData);
+    } catch (error) {
```
same

TarikVelic committed:
FEATURE/TariikMVP/Update

--------------------------------------------------------------------
 MAIN
 * Small changes in CustomTable.scss
 * Removed catch from AddProjectsModal.js
--------------------------------------------------------------------
 OTHER
 * <other msg>
--------------------------------------------------------------------

TarikVelic committed:
Merge branch 'develop' into FEATURE/TariikMVP/Update
